server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -80,8 +80,22 @@ class InteractiveSession(
 
     def mergeConfList(list: Seq[String], key: String): Unit = {
       if (list.nonEmpty) {
-        val newList = (list ++ conf.get(key)).mkString(",")
-        builder.setConf(key, newList)
+        val oldList = builder.getConf(key)
```

Comment:
You don't need to expose `getConf` in the public API for this. There's a `conf` map above in this method with all the user configuration. Just modify it before creating the client.